File: daily/203. 移除链表元素.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} head
 * @param {number} val
 * @return {ListNode}
 */
var removeElements = function(head, val) {
    var p=new ListNode,temp;
    p.next=head,temp=p;
    while(temp.next){
        if(temp.next.val==val){
            temp.next=temp.next.next;
        }
        else    
            temp=temp.next;
    }
    return p.next;
};

File: daily/C++版本/203. 移除链表元素.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* removeElements(ListNode* head, int val) {
       ListNode *p=new ListNode,*temp;
       p->next=head,temp=p;
       while(temp->next){
           if(temp->next->val==val)
                temp->next=temp->next->next;
            else
                temp=temp->next;
       }
        return p->next;   
    }
};
